### PR TITLE
[SPARK-11352] [SQL] Escape */ in the generated comments.

### PR DESCRIPTION
**Pull request by yhuai**
https://issues.apache.org/jira/browse/SPARK-11352

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Expression.scala [hunk 1]
```diff
@@ -95,7 +95,7 @@ abstract class Expression extends TreeNode[Expression] {
     ctx.subExprEliminationExprs.get(this).map { subExprState =>
       // This expression is repeated meaning the code to evaluated has already been added
       // as a function and called in advance. Just use it.
-      val code = s"/* $this */"
+      val code = s"/* ${this.toString.replace("*/", "\\*\\/")} */"
```
Comment (nongli) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Expression.scala]: Should we put this as a utility in ctx?

Maybe. def escapeComment(< thing to put inside /* */>)

This seems easy to get wrong in the future too.

Comment (yhuai) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Expression.scala]: Yeah. Let me do that.

**Comment by rxin** [comment]
We should backport this into 1.5 as well.

**Comment by yhuai** [comment]
sure

**Comment by nongli** [comment]
LGTM

**Comment by yhuai** [comment]
Merging to master, branch 1.6, and branch 1.5.

**Comment by yhuai** [comment]
I will have a pr for branch 1.5. The current on conflicts a lot with branch 1.5.

**Pull request closed without merging**